src/trusted/service_runtime/nacl_syscall_common.c

```diff
@@ -2085,214 +1509,57 @@ int32_t NaClSysMmapIntern(struct NaClApp        *nap,
-                                            &metadata,
-                                            nap->validation_cache),
-                                 NaClValidationFailed);
-      NaClLog(3, "NaClSysMmap: prot_exec, validator_status %d\n",
-              validator_status);
-      NaClMetadataDtor(&metadata);
-
-      if (NaClValidationSucceeded == validator_status) {
-        /*
-         * Check if target address range is actually available.  It
-         * must be neither in use by NaClText interface, nor used by
-         * previous mmap'd code.  We record mmap'd code regions in the
-         * NaClText's data structures to avoid lo both having to deal
-         * with looking in two data structures.  We could do this
-         * first since this is a cheaper check, but it shouldn't
-         * matter since application errors ought to be rare and we
-         * shouldn't optimize for error handling, and this makes the
-         * code simpler (releasing a created region is more code).
-         */
-        NaClXMutexLock(&nap->dynamic_load_mutex);
-        //We don't check prot in mmap
-        ret = NaClDynamicRegionCreate(nap, NaClUserToSys(nap, usraddr), length,
-                                      1);
-        NaClXMutexUnlock(&nap->dynamic_load_mutex);
-        if (!ret) {
-          NaClLog(3, "NaClSysMmap: PROT_EXEC region"
-                  " overlaps other dynamic code\n");
-          map_result = -NACL_ABI_EINVAL;
-          goto cleanup;
-        }
-        /*
-         * Remove scratch mapping.
-         */
-        NaClDescUnmapUnsafe(ndp, (void *) image_sys_addr, length);
-        /*
-         * We must succeed in mapping into the untrusted executable
-         * space, since otherwise it would mean that the temporary
-         * hole (for Windows) was filled by some other thread, and
-         * that's unrecoverable.  For Linux and OSX, this should never
-         * happen, since it's an atomic overmap.
-         */
-        NaClLog(3, "NaClSysMmap: mapping into executable memory\n");
-        image_sys_addr = (*NACL_VTBL(NaClDesc, ndp)->
-                          Map)(ndp,
-                               nap->effp,
-                               (void *) sysaddr,
-                               length,
-                               NACL_ABI_PROT_READ | NACL_ABI_PROT_EXEC,
-                               NACL_ABI_MAP_PRIVATE | NACL_ABI_MAP_FIXED,
-                               offset);
-        if (image_sys_addr != sysaddr) {
-          NaClLog(LOG_FATAL,
-                  "NaClSysMmap: map into executable memory failed:"
-                  " got 0x%"NACL_PRIxPTR"\n", image_sys_addr);
-        }
-        map_result = (int32_t) usraddr;
-        goto cleanup;
-      }
 
-      NaClLog(3,
-              "NaClSysMmap: did not validate in readonly_text mode;"
-              " attempting to use dyncode interface.\n");
 
-      if (holding_app_lock) {
-        NaClVmHoleClosingMu(nap);
```

Comment:
I'm not fully sure what this does, please explain so we can make sure this won't cause any vulnerabilities/race conditions